src/app/portlets/dot-edit-content/dot-edit-content.component.ts

```diff
@@ -137,58 +130,66 @@ export class DotEditContentComponent implements OnInit {
     }
 
     /**
-     * Handle the page lock state
-     *
-     * @param {boolean} locked
+     * Handle the iframe page load
+     * @param {any} $event
      * @memberof DotEditContentComponent
      */
-    lockPageHandler(locked: boolean): void {
-        this.dotGlobalMessageService.display(
-            this.dotMessageService.get(locked ? 'dot.common.message.locking' : 'dot.common.message.unlocking')
-        );
-        this.pageViewService.lock(this.pageIdentifier).subscribe((res) => {
-            this.dotGlobalMessageService.display(
-                this.dotMessageService.get(locked ? 'dot.common.message.locked' : 'dot.common.message.unlocked')
-            );
-        });
+    onLoad($event): void {
+        this.dotLoadingIndicatorService.hide();
     }
 
     /**
-     * Handle the state of the pate
+     * Handle the changes of the state pf the page
      *
-     * @param {string} state
+     * @param {*} state
      * @memberof DotEditContentComponent
      */
-    statePageHandler(state: PageState): void {
-        // TODO: do the request and set the new state
-        // this.editPageService.get(this.pageUrl, state);
+    statePageHandler(state: DotEditPageState): void {
+        if (state.lock !== null) {
+            this.dotGlobalMessageService.display(this.dotMessageService.get('dot.common.message.saving'));
+        }
+
+        this.editPageService.setPageState(this.page, state).subscribe((dotRenderedPageState: DotRenderedPageState) => {
+            if (dotRenderedPageState.dotRenderedPage) {
+                this.setPage(dotRenderedPageState.dotRenderedPage);
+            }
+
+            if (state.lock !== null) {
+                this.dotGlobalMessageService.display(this.dotMessageService.get('dot.common.message.saved'));
```

Comment:
We should show something like 'locked' insted

Reply:
+1 locked sounds more meningful

Reply:
Agreed, done.